olp-cpp-sdk-core/tests/olpclient/unittest/OlpClientHttpTest.cpp

```diff
@@ -0,0 +1,1213 @@
+/*
+ * Copyright (C) 2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+#include <gmock/gmock.h>
+#include <gtest/gtest.h>
+
+#include <chrono>
+#include <future>
+#include <string>
+
+#include <olp/core/client/OlpClient.h>
+#include <olp/core/client/OlpClientFactory.h>
+#include <olp/core/client/OlpClientSettingsFactory.h>
+
+#include <olp/core/http/Network.h>
+
+#include "NetworkAsyncHandlerImpl.h"
+
+using ::testing::_;
+
+class NetworkMock : public olp::http::Network {
+ public:
+  MOCK_METHOD(olp::http::SendOutcome, Send,
+              (olp::http::NetworkRequest request,
+               olp::http::Network::Payload payload,
+               olp::http::Network::Callback callback,
+               olp::http::Network::HeaderCallback header_callback,
+               olp::http::Network::DataCallback data_callback),
+              (override));
+
+  MOCK_METHOD(void, Cancel, (olp::http::RequestId id), (override));
+};
+
+class ClientHttp : public ::testing::TestWithParam<bool> {
+ protected:
+  olp::client::OlpClientSettings client_settings_;
+  olp::client::OlpClient client_;
+};
+
+INSTANTIATE_TEST_SUITE_P(TestOffline, ClientHttp, ::testing::Values(false));
```

Comment:
This, to be honest, I never saw before. Value-parameterized test suite with only one value. Makes no sense at all. Why even bother?

Reply:
OlpClientTest.cpp:52

That I suppose should help to filter out online & offline tests. 

OlpClientTest & OlpClientHttpTest are compiled in the same binary. I've made new tests part of the same test suite.

Reply:
Filtering you can do by --gtest_filter=-*Online* as it is done now. Doesn't matter if it is in the test fixture name or in the test case name.